scripts/src/logging.ts

```diff
@@ -0,0 +1,72 @@
+import * as winston from "winston";
+import {GenericTextTransportOptions} from "winston";
+import {GenericTransportOptions} from "winston";
+import {FileTransportOptions} from "winston";
+
+export interface LogTarget {
+	name: string;
+	type: "console" | "file";
+	level: "debug" | "info" | "warn" | "error";
+	timestamp?: boolean | (() => string | boolean);
+}
+
+export interface ConsoleTarget extends LogTarget {
+	type: "console";
+	name: "console";
+}
+
+export interface FileTarget extends LogTarget {
+	type: "file";
+	file: string;
+}
+
+export function getLogger(...targets: LogTarget[]): winston.LoggerInstance {
+	const options: winston.LoggerOptions = {};
+	options.transports = targets.map(target => createTarget(target));
+
+	return new winston.Logger(options);
+}
+
+type WinstonTransportOptions = GenericTransportOptions & GenericTextTransportOptions;
+function createTarget(target: LogTarget): winston.TransportInstance {
+	let cls: { new (options: WinstonTransportOptions): winston.TransportInstance };
```

Comment:
I don't know what this means

Reply:
`cls` is a class, that is, a constructor function which expects one param (`options`) and returns a new instance of `winston.TransportInstance`.
notice that later there's:
`new cls(options)`